Automattic/wp-calypso
# Gutenboarding: Update the domains picker to make domain upgrades available

This PR re-introduce paid domains in the gutenboarding domain picker, linking them to a sub-flow that actually ask the user to create an account, creates the site, and send them through the calypso checkout so they can pay for a personal plan before they are redirected back to the design section of gutenboarding. 

How to test:
--------------

1. Go to http://calypso.localhost:3000/gutenboarding (or https://calypso.live/gutenboarding?branch=update/gutenboarding-domains-flow )
1. Select a vertical and enter a site title: you should get a domain showing up in the top bar
1. Click it and choose one of the paid domains in the list (or look for a new term and choose one) 
1. You should see a modal dialog telling you what happens next. Make sure you can close it by clicking 'do it later' or clicking outside of the modal
  ![image](https://user-images.githubusercontent.com/1554855/76072460-4d040c00-5f98-11ea-9964-e1f613230726.png)
1. Choose a paid domain again, and this time click on the "create your site and..." button
1. (optional). If you are unlogged, you should get a new dialog asking you to enter your email to create an account. Create one (trick: where we are going you don't need to verify your email, so you can totally make up a non-existent email address)
1. You should see a placeholder-full screen telling you your site is being created. 
1. You should land on the checkout page, with a personal plan already on the cart. Purchase it (give your new fake account some credits first, if that's the case)
  ![image](https://user-images.githubusercontent.com/1554855/76072507-63aa6300-5f98-11ea-962a-5420c43001ec.png)
1. You should land on /gutenboarding/design. The new 'paid' domain should be selected in the domains component in the top bar
